### PR TITLE
fix: delete welcome and freeform post

### PR DESCRIPTION
**Pull request by sshanzel**
Deleting of posts for Welcome and Freeform post should be allowed.

### CODE REVIEW

**Review by capJavert — CHANGES_REQUESTED** [review]
👌 Nice

We probably need to update tests? 

Also remove mention of shared post and createSharedPost helper (since it no longer applies to only that one).

**Review by capJavert — CHANGES_REQUESTED** [review]
👌 Nice

We probably need to update tests? 

Also remove mention of shared post and createSharedPost helper (since it no longer applies to only that one).

**Comment by sshanzel** [comment]
> We probably need to update tests?
>
> Also remove mention of shared post and createSharedPost helper (since it no longer applies to only that one).

Right, we need to add tests for this. Though for the helper, I think we can keep it as it is used by most of the tests related to deleting which would work the same for other types.

**Comment by capJavert** [comment]
> > We probably need to update tests?
> > Also remove mention of shared post and createSharedPost helper (since it no longer applies to only that one).
> 
> Right, we need to add tests for this. Though for the helper, I think we can keep it as it is used by most of the tests related to deleting which would work the same for other types.

Agreed, we can just rename it maybe 👍

**Comment by sshanzel** [comment]
I think we don't even have to touch it, we won't just use it on the new tests that we will be creating since it won't be useful.

File: src/schema/posts.ts [hunk 1]
```diff
@@ -980,7 +981,7 @@ export const resolvers: IResolvers<any, Context> = {
       await ctx.con.transaction(async (manager) => {
         const repo = manager.getRepository(Post);
         const post = await repo.findOneBy({ id });
-        if (post.type === PostType.Share) {
+        if (squadPostTypes.includes(post.type)) {
```
Comment (idoshamun) [inline src/schema/posts.ts:984]: I think we should not check the type here even.
If user has access to delete a post within squad, why does the type matter?

Comment (sshanzel) [reply src/schema/posts.ts:984]: Makes sense. I think this was just an additional check for `article` types.

**Review by idoshamun — APPROVED** [review]

**Comment by sshanzel** [comment]
@capJavert let me know what you think 🥳

**Review by capJavert — APPROVED** [review]

**Pull request merged**